packages/material-ui/src/Radio/Radio.js

```diff
@@ -67,7 +67,7 @@ const Radio = React.forwardRef(function Radio(props, ref) {
     size = 'medium',
     ...other
   } = props;
-  const radioGroup = React.useContext(RadioGroupContext);
+  const radioGroup = useRadioGroup();
```

Comment:
I'm happy to revert these changes if we want to avoid doing this for bundle size reasons.

Reply:
I prefer this for readability and drive-by integration testing.